### ISSUE TITLE
Prototype: Ambassador Authentication Filter

### ISSUE THREAD

**Issue by plombardi89** [datawire/ambassador#59]
First stage of GH-9

- Prototype code to validate the external authentication mechanism and learn how to implement properly as well as learn Envoy toolchain.
- Prototype Documentation
- Plan for moving from Prototype to GA.

**Comment by kflynn** [comment]
There's now a separate ticket for docs. We need to have CI pushing this image to Docker before we can close it though.

**Comment by ark3** [comment]
CI is pushing to Docker for ambassador-envoy and for ambassador-auth-service
